source/brailleDisplayDrivers/seikantk.py

```diff
@@ -0,0 +1,264 @@
+		self._dev = hwIo.Hid(path=self.path, onReceive=self._onReceive)
+		if self._dev._file == INVALID_HANDLE_VALUE:
+			raise RuntimeError("No MINI-SEIKA display found, open error")
+		self._dev.setFeature(SEIKA_CONFIG)  # baudrate, stopbit usw
+		self._dev.setFeature(SEIKA_CMD_ON)  # device on
+		self._dev.write(SEIKA_REQUEST_INFO)  # Request the Info from the device
+
+		# wait and try to get info from the Braille display
+		for i in range(30):  # the info-block is about
+			self._dev.waitForRead(TIMEOUT)
+			if self.numCells:
+				log.info("Seikanotetaker an USB-HID, Cells {c} Buttons {b}".format(c=self.numCells, b=self.numBtns))
+				break
+
+		if self.numCells == 0:
+			self._dev.close()
+			raise RuntimeError("No MINI-SEIKA display found, no response")
+
+	def terminate(self):
+		try:
+			super(BrailleDisplayDriver, self).terminate()
+		finally:
+			self._dev.close()
+
+	def display(self, cells: List[int]):
+		# cells will already be padded up to numCells.
+		cellBytes = SEIKA_SEND_TEXT + self.numCells.to_bytes(1, 'little') + bytes(cells)
+		self._dev.write(cellBytes)
+
+	def _onReceive(self, data: bytes):
+		stream = BytesIO(data)
+		cmd = stream.read(3)
+		self._hidBuffer += cmd[1:2]
+		if len(self._hidBuffer) == 3:
+			self.status = 1
+		elif len(self._hidBuffer) == 4:
+			self.cmdlen = cmd[1]
+			self.status = 2
+		elif self.status == 2 and len(self._hidBuffer) == self.cmdlen + 4:
+			command = self._hidBuffer[0:3]
+			arg = self._hidBuffer[3:self.cmdlen + 4]
+			self.status = 0
+			self._hidBuffer = b""
+			
+			if command == SEIKA_INFO:
+				self._handInfo(arg)
+			elif command == SEIKA_ROUTING:
+				self._handRouting(arg)
+			elif command == SEIKA_KEYS:
+				self._handKeys(arg)
+			elif command == SEIKA_KEYS_ROU:
+				self._handKeysRouting(arg)
+			else:
+				log.debug("other data.")
+
+	def _handInfo(self, arg: bytes):
+		self.numCells = arg[2]
+		self.numBtns = arg[1]
+
+	def _handRouting(self, arg: bytes):
+		Rou = 0
+		for i in range(arg[0]):
+			for j in range(8):
+				if arg[i + 1] & (1 << j):
+					Rou = i * 8 + j
+					
+					gesture = InputGestureRouting(Rou)
+					try:
+						inputCore.manager.executeGesture(gesture)
+					except inputCore.NoInputGestureAction:
+						log.debug("No Action for routing command")
+						pass
+
+	def _handKeys(self, arg: bytes):
+		Brl = arg[1]
+		Key = arg[2] | (arg[3] << 8)
+		Btn = 0  # Seika has no button
+		if not (Key or Btn or Brl):
+			pass
+		if Key:  # Mini Seika has 2 Top and 4 Front ....
+			gesture = InputGesture(keys=Key)
+		if Btn:  # Mini Seika has no Btn ....
+			gesture = InputGesture(keys=Btn)
+		if Brl:  # or how to handle Brailleinput?
+			gesture = InputGesture(dots=Brl)
+		if Key or Btn or Brl:
+			try:
+				inputCore.manager.executeGesture(gesture)
+			except inputCore.NoInputGestureAction:
+				log.debug("No Action for keys ")
+				pass
+
+	def _handKeysRouting(self, arg: bytes):
+		argk = b"\x03" + arg[1:]
```

Comment:
It would be good to know the meaning of `b"\x03"` by turning it to a named constant.